src/Tools/TestDiscoveryWorker/Program.cs

```diff
@@ -24,76 +24,85 @@
     return ExitFailure;
 }
 
-using var pipeClient = new AnonymousPipeClientStream(PipeDirection.In, args[0]);
-using var sr = new StreamReader(pipeClient);
-string? output;
-
-// Wait for 'sync message' from the server.
-do
+try
```

Comment:
similar change to ensure exception details are captured if something goes wrong.